### PR TITLE
feat: support local names in custom name section

### PR DESCRIPTION
**Pull request by ColinEberhardt**
This completes the custom name section implementation, making it easier to achieve parity between wast and wasm parsers

### CODE REVIEW

**Comment by xtuc** [comment]
Just FYI custom section in WAST is not possible yet, but there's a discussion going on to add them.

File: packages/ast/src/index.js [hunk 1]
```diff
@@ -44,18 +44,34 @@ export function functionNameMetadata(
...
 ): ModuleMetadata {
   const n: ModuleMetadata = {
     type: "ModuleMetadata",
     sections
   };
-
   if (functionNames.length) {
     n.functionNames = functionNames;
   }
+  if (localNames.length) {
```
Comment (xtuc) [inline packages/ast/src/index.js]: While I understand this, I would prefer something more explicit: `localNames.length > 0`.

File: packages/wasm-parser/src/decoder.js [hunk 2]
```diff
@@ -1006,11 +1006,37 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
 
   function parseNameSectionLocals() {
     const subSectionSizeInBytesu32 = readU32();
-    const subSectionSizeInBytes = subSectionSizeInBytesu32.value;
     eatBytes(subSectionSizeInBytesu32.nextIndex);
 
-    // TODO: add support for naming locals - for now we just eat this section
-    eatBytes(subSectionSizeInBytes);
+    const numbeOfFunctionsu32 = readU32();
```
Comment (xtuc) [inline packages/wasm-parser/src/decoder.js]: Nit: 

```diff
- numbeOfFunctionsu32
+ numberOfFunctionsu32
```

**Comment by ColinEberhardt** [comment]
Thanks @xtuc I've addressed those comments

**Pull request merged**